src/compiler/ast/common/Node.ts

```diff
@@ -1124,20 +1124,22 @@ export class Node<NodeType extends ts.Node = ts.Node> {
      * Returns undefined if the initial parent doesn't match the condition.
      * @param condition - Condition that tests the parent to see if the expression is true.
      */
-    getParentWhile<T extends Node>(condition: (node: Node) => node is T): T | undefined;
+    getParentWhile<T extends Node>(condition: (parent: Node, node: Node) => parent is T): T | undefined;
     /**
      * Goes up the parents (ancestors) of the node while a condition is true.
      * Returns undefined if the initial parent doesn't match the condition.
      * @param condition - Condition that tests the parent to see if the expression is true.
      */
-    getParentWhile(condition: (node: Node) => boolean): Node | undefined;
-    getParentWhile(condition: (node: Node) => boolean) {
-        let node: Node | undefined = undefined;
-        let nextParent: Node | undefined = this.getParent();
-        while (nextParent != null && condition(nextParent)) {
-            node = nextParent;
-            nextParent = nextParent.getParent();
-        }
+    getParentWhile(condition: (parent: Node, node: Node) => boolean): Node | undefined;
+    getParentWhile(condition: (parent: Node, node: Node) => boolean) {
+        let node: Node | undefined;
```

Comment:
Bit of a nitpick, but I think you could assign `this` to `node` here then keep using a `while` loop?